src/Microsoft.Azure.SignalR/EndpointRouters/DefaultEndpointRouter.cs

```diff
@@ -69,7 +80,7 @@ private ServiceEndpoint GetEndpointAccordingToWeight(ServiceEndpoint[] available
                 var remain = endpointMetrics.ConnectionCapacity -
                              (endpointMetrics.ClientConnectionCount +
                               endpointMetrics.ServerConnectionCount);
-                var weight = remain > 0 ? remain : 1;
+                var weight = Math.Max((int)((double)remain / endpointMetrics.ConnectionCapacity * 1000), 1);
```

Comment:
minor: we can keep it in integer operations:
`remain * 1000 / endpointMetrics.ConnectionCapacity`
unit 1000 * 1000 connections/unit * 1000 = 1G, it is less than max value of integer (2G).